File: ios/Plugin/NavigationBarPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(NavigationBarPlugin, "NavigationBar",
           CAP_PLUGIN_METHOD(setBackgroundColor, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setNavigationBarDividerColor, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setNavigationBarContrastEnforced, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setStyle, CAPPluginReturnPromise);
)

File: ios/Plugin/NavigationBarPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NavigationBarPlugin)
public class NavigationBarPlugin: CAPPlugin {
    @objc func setBackgroundColor(_ call: CAPPluginCall) -> Void {
        call.resolve();
    }

    @objc func setNavigationBarDividerColor(_ call: CAPPluginCall) -> Void {
        call.resolve();
    }

    @objc func setNavigationBarContrastEnforced(_ call: CAPPluginCall) -> Void {
        call.resolve();
    }

    @objc func setStyle(_ call: CAPPluginCall) -> Void {
        call.resolve();
    }
}
